Stratis.Bitcoin/BlockPulling/BlockPuller.cs

```diff
@@ -22,14 +36,21 @@ public class DownloadedBlock
 
 		public class BlockPullerBehavior : NodeBehavior 
 		{
+
+			private readonly CancellationTokenSource cancellationToken = new CancellationTokenSource();
```

Comment:
I think we should not instantiate params on creation but in the constructor, it makes it really hard to debug that way.